hack/update-cluster

```diff
@@ -0,0 +1,81 @@
+#!/bin/sh
+set -e
+
+# Configuration
+#
+# Path to kubeconfig
+KUBECONFIG=${KUBECONFIG:-cluster/auth/kubeconfig}
+# Version of Kubernetes to upgrade to.
+NEW_VERSION=${NEW_VERSION:-"v1.3.0-beta.2_coreos.0"}
+
+# Shortcut for kubectl command
+kc="kubectl --kubeconfig=${KUBECONFIG} --namespace=kube-system"
+# Temp dir for manifests
+temp=$(mktemp -d -t k8s-upgrade-$NEW_VERSION.XXX)
+# New image to deploy.
+new_image=${new_image:-"quay.io\/coreos\/hyperkube:${NEW_VERSION}"}
+
+update_image() {
+  local filepath=$1
+  local newfile=$(echo $filepath | sed "s/\.yaml/-${NEW_VERSION}.yaml/")
+  echo "newfile: " $newfile
+  sed "s/image:.*hyperkube.*$/image: ${new_image}/g" "${filepath}" > "${newfile}"
+}
+
+# Update kube-apiserver
+# Object is a DaemonSet, so we must first delete, then create to update.
+echo
+echo "Updating apiserver"
+$kc get ds kube-apiserver -oyaml > "${temp}/kube-apiserver.yaml"
+update_image "${temp}/kube-apiserver.yaml"
+# Delete kube-apiserver and wait for temp-apiserver to take over.
+$kc delete --timeout=1s -f "${temp}/kube-apiserver.yaml"
+while true; do
+        echo "checking for apiserver..."
```

Comment:
Yeah, probably not accomplishing much here, we might wanna wait or something.

The original idea is basically that since we're explicitly bringing down the API server, let's hang out for a bit until the checkpointed API server kicks in and we can continue rolling the rest of the cluster (API server will sort itself out).

So instead of a timeout or something when creating the new daemonset we just wait until we have an API server to talk to and then continue.